_posts/2019-08-12-what-the-cluster.markdown

```diff
@@ -0,0 +1,75 @@
+* You can share your tables and materialized views with consumers who do not have a snowflake account by using reader accounts.
+
+
+One of the reoccurring themes was clusters. Snowflake uses the term cluster a lot!
+
+### Data Cluster
+
+This is how data is ordered and sorted in micro-partitions. Snowflake will cluster your data automatically into micro-partitions to allow for faster retrieval of frequently requested data.Micro-partitions is Snowflakes unique way of storing large amounts of data in a way that enables fast retrieval of frequently accessed data. Micro-partitions are immutable , compressed and encrypted at rest.
+You can evaluate how your table is clustered by looking at certain columns that your users will interface with.
+
+```sql
+SELECT
+SYSTEM$CLUSTERING_INFORMATION('UNICORNS','(SIZE,AGE)');
+```
+
+### Compute Cluster
+
+A compute cluster is a collection of one or more VMs connected within a mesh. All virtual warehouses are an independent compute cluster which does not share its resources with other compute clusters. As each compute cluster does not share compute resources all other virtual warehouses will have no impact on eachother.As a standard for your warehouse you will only have a single compute cluster meaning you are unable to scale out, if you need more than one compute cluster you would need to utilise the multi cluster warehouses.
+
+### Cluster Keys
+
+Used to help keep all frequently accessed data in the same micro-partition for faster retrieval. This is only ideal for very large tables as snowflake automatically clusters your data into micro-partitions . Using cluster keys will override snowflakes natural clustering.
+So say you want to find unicorns; you want to know how old they are, how big they are and where they were last spotted. This is the information you need to go forth and find unicorns so you want this information to be stored in the same micro-partition so you can get hold of this vital information fast.
+Cluster keys are used to keep frequently accessed data in the same micro-partition.So if I wanted to re cluster my UNICORNS table from above I could set clustering keys on AGE,SIZE and LOCATION.
+
+```sql
+ALTER TABLE UNICORNS CLUSTER BY (AGE,SIZE,LOCATION);
+ALTER TABLE UNICORNS RECLUSTER;
+```
+
+### Multi Cluster Warehouses
+
+Multi cluster warehouses can automatically scale out and spin up more compute clusters based on user concurrency needs. With multi cluster warehouses you can configure the minimum and maximum number of server clusters up to a maximum of ten. Snowflake documentation is really good and you can find out how much multi cluster warehouses will cost you in credits in different scenarios [here:](https://docs.snowflake.net/manuals/user-guide/warehouses-multicluster.html)
```

Comment:
The link with `[here:] looks strange:
![image](https://user-images.githubusercontent.com/8016590/63477958-401b4180-c4ba-11e9-9530-926b60431568.png)

Did you mean a full stop?